nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processors/src/main/java/org/apache/nifi/processors/morphlines/ImplementMorphlines.java

```diff
@@ -0,0 +1,219 @@
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for success.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Relationship for failure of morphlines.")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+            .add(MORPHLINES_FILE)
+            .add(MORPHLINES_ID)
+            .add(MORPHLINES_OUTPUT_FIELD)
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = ImmutableSet.<Relationship>builder()
+            .add(REL_SUCCESS)
+            .add(REL_FAILURE)
+            .build();
+
+
+    private File morphLinesFile;
+    private String morphLinesId;
+    private String morphlinesOutputField;
+    private PropertyValue morphLinesFileProperty;
+    private PropertyValue morphLinesIdProperty;
+    private PropertyValue morphLinesOutputField;
+    private MorphlineContext morphlineContext;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) throws Exception {
+	morphlinesFileProperty = context.getProperty(MORPHLINES_FILE);
+	morphlinesIdProperty = context.getProperty(MORPHLINES_ID);
+	morphlinesOutputFieldProperty = context.getProperty(MORPHLINES_OUTPUT_FIELD);
+        morphlineContext = new MorphlineContext.Builder().build();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+	morphLinesFile = new File(morphlinesFileProperty.evaluateAttributeExpressions(flowfile).getValue());
+	morphLinesId = morphlinesIdProperty.evaluateAttributeExpressions(flowfile).getValue();
+	morphlinesOutputField = morphlinesOutputFieldProperty.evaluateAttributeExpressions(flowfile).getValue();
+
+        final AtomicLong written = new AtomicLong(0L);
+        final byte[] value = new byte[(int) flowFile.getSize()];
+
+        try{
+            flowFile = session.write(flowFile, new StreamCallback() {
+                @Override
+                public void process(InputStream in, OutputStream out) throws IOException {
+                    StreamUtils.fillBuffer(in, value);
+                    Record record = new Record();
+                    record.put(Fields.ATTACHMENT_BODY, value);
+                    Collector collectorRecord = new Collector();
+                    morphline = new org.kitesdk.morphline.base.Compiler().compile(morphLinesFile, morphLinesId, morphlineContext, collectorRecord);
```

Comment:
You said before that you couldn't move this out of the onTrigger due to needing the collectorRecord to create and compile it.

According to the Cloudera intro documentation "Example Driver Program". It creates the compiler then processes each record. Can't we do the same here? Relevant section below:

```
  MorphlineContext context = new MorphlineContext.Builder().build();
  Command morphline = new Compiler().compile(configFile, null, context, null);

  // process each input data file
  Notifications.notifyBeginTransaction(morphline);
  for (int i = 1; i < args.length; i++) {
    InputStream in = new FileInputStream(new File(args[i]));
    Record record = new Record();
    record.put(Fields.ATTACHMENT_BODY, in);
    morphline.process(record);
    in.close();
  }
```



[1] http://blog.cloudera.com/blog/2013/07/morphlines-the-easy-way-to-build-and-integrate-etl-apps-for-apache-hadoop/